dnf/module/repo_module_version.py

```diff
@@ -0,0 +1,203 @@
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Library General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+import hawkey
+
+import dnf
+from dnf.module.exceptions import NoProfileException, PossibleProfilesExceptions, \
+    NoProfilesException
+from dnf.subject import Subject
+
+
+class RepoModuleVersion(object):
+    def __init__(self, module_metadata, base, repo):
+        self.module_metadata = module_metadata
+        self.repo = repo
+        self.base = base
+        self.parent = None
+        self.repo_module = None
+
+    def __lt__(self, other):
+        # for finding latest
+        assert self.full_stream == other.full_stream
+        if self.repo_module.conf.locked._get():
+            return self.version < self.repo_module.conf.version._get()
+        return self.module_metadata.get_version() < other.module_metadata.get_version()
+
+    def __repr__(self):
+        return self.full_version
+
+    def report_profile_error(self, profile, default_profiles_used=False):
+        if default_profiles_used:
+            raise NoProfileException("{}/{}".format(self.full_version, profile))
+        elif self.profiles:
+            raise PossibleProfilesExceptions("{}/{}".format(self.full_version, profile),
+                                             self.profiles)
+        else:
+            raise NoProfilesException("{}/{}".format(self.full_version, profile))
+
+    def install(self, profiles, default_profiles):
+        result = self._install_profiles(profiles, False)
+        if not profiles:
+            result |= self._install_profiles(default_profiles, True)
+
+        profiles.extend(list(self.repo_module.conf.profiles._get()))
+        profiles.extend(default_profiles)
+        self.base._module_persistor.set_data(self.repo_module, stream=self.stream,
+                                             version=self.version, profiles=sorted(set(profiles)))
+
+        return result
+
+    def _install_profiles(self, profiles, defaults_used):
+        installed = self.base.sack.query().installed().run()
+        installed_nevras = [str(pkg) for pkg in installed]
+
+        result = False
+        for profile in profiles:
+            if profile not in self.profiles:
+                self.report_profile_error(profile, defaults_used)
+
+            for nevra_object in self.profile_nevra_objects(profile):
+                nevr = self.nevra_object_to_nevr_str(nevra_object)
+                nevra = "{}.{}".format(nevr, nevra_object.arch)
+
+                if nevra not in installed_nevras:
+                    self.base.install(nevr, reponame=self.repo.id, forms=hawkey.FORM_NEVR)
```

Comment:
Strict option not inherited. Please lets make life of user more flexible and allow --skip-broken option.

Reply:
Are you sure that reponame=self.repo.id is correct? For example if update in in upgrade repo but some part of rpm is still in Fedora repo